### PR TITLE
Create widget for defining CensusAreas

### LINKED ISSUE #101: Create geographic widget for selecting CensusAreas
Create a form widget allowing a user to select a group of census tracts or blocks to create a CensusArea. The widget should:

- Display all available Census blockgroups on a map
- Allow the user to select, deselect, and clear multiple blockgroups from the map
- Save the blockgroups to `CensusArea.fips_codes`

We built out some prior art for this kind of "geographic multiselect" in EDI. See: 
- https://github.com/datamade/erikson-edi/blob/ad68b7853b967be3e1adc3cf6330b8c5e63218fb/edi/forms.py#L59-L68
- https://github.com/datamade/erikson-edi/blob/ad68b7853b967be3e1adc3cf6330b8c5e63218fb/edi/widgets.py#L9-L53

### LINKED ISSUE #158: Make ACS data imports resistant to updates
Right now, the import_data command truncates the CensusObservation table and bulk-imports the CSV output from the Makefile. We should modify this to make it iterate the CSV, check if a CensusObservation exists matching the FIPS code and variable name, and optionally update that record.

It would also be useful to add a --truncate flag to the command to allow us to truncate the table and start over if it gets into a weird state.

### PR DESCRIPTION
**Pull request by jeancochrane**
## Overview

Closes #101, closes #158 

Define a `MultiSelectGeometryWidget` for defining CensusAreas using Census block groups.

In addition, update the import pipeline to:

1. Import Census block group shapefiles for Philadelphia
2. Make the import of ACS data accommodate data refreshes

### Setup

- Run migrations

### Test the Makefile changes

- Remove the derived shapefile: `rm data/final/shapefiles/cb_2018_42_bg_500k.shp`
- Change to the data directory: `cd data`
- Recreate the file: `docker-compose run --rm make final/shapefiles/cb_2018_42_bg_500k.shp`

### Test the data import

- Run `python manage.py import_data` and confirm that:
    - Your existing ACS data gets updated, not deleted (the script should print to the console indicating this)
    - The script successfully imports blockgroup data

### Test the multiselect widget

- Run the development server
- Navigate to http://localhost:8000/census-areas/create/ and try creating CensusAreas:
    - Create a CensusArea with one block group
    - Create a CensusArea with multiple block groups
    - Try to create a CensusArea with no block groups, and confirm that the form shows you an error
    - After creating CensusAreas, view their "Edit" pages and confirm that the selected block groups reappear on the map
- Navigate to the DAD for an ACS-compatible survey and try using your new CensusAreas to compare data
    - Confirm that the bars look different across CensusAreas

### CODE REVIEW

**Comment by beamalsky** [comment]
When I run `docker-compose run --rm make final/shapefiles/cb_2018_42_bg_500k.shp` I get this error: 

`ERROR: Couldn't find env file: /Users/beamalsky/Desktop/just-spaces/data/.env`

Looks like I have `.env.example` but no `.env` file. Do I need to generate my own `CENSUS_API_KEY`?

**Comment by beamalsky** [comment]
To try to fix this I ran `blackbox_cat configs/.env.gpg > data/.env` after `rm data/final/shapefiles/cb_2018_42_bg_500k.shp` in the steps above.

File: surveys/templates/partials/multiselect_geometry_widget.html [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+      center: [40, -75.16],  // Centroid of Philadelphia
+      zoom: 11,
+      dragging: true,
+      touchZoom: true,
+      zoomControl: true,
+      tap: true,
+      scrollWheelZoom: false
+    });
+
+    var tileUrl = 'https://{s}.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}{r}.png';
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/templates/partials/multiselect_geometry_widget.html]: Other Leaflet settings: https://github.com/datamade/just-spaces/blob/bd4396e0a6b7cd311c7c8a25f8c4677440386447/justspaces/settings.py#L192

[the remark this answers is not in the document]
Comment (jeancochrane) [reply surveys/templates/partials/multiselect_geometry_widget.html]: Hmm, how is `LEAFLET_CONFIG` being used on the site? Is it imported anywhere, or is it just documenting examples? I couldn't find any uses of it when searching the code.

File: README.md [hunk 2]
```diff
@@ -87,5 +88,15 @@
 
     Navigate to http://localhost:8000/.
 
+## Importing new ACS data
+
+If you'd like to refresh ACS data, there are a few steps you'll need to take:
+
+0. If your database is running in production, make a backup of it with `pg_dump` just in case
```
[the remark this answers is not in the document]
Comment (beamalsky) [reply README.md]: That sounds good!

File: surveys/templates/partials/multiselect_geometry_widget.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/templates/partials/multiselect_geometry_widget.html]: It's part of `django-leaflet`: https://django-leaflet.readthedocs.io/en/latest/templates.html?highlight=LEAFLET_CONFIG#configuration

To document what I said IRL, I like the look of your map but I think some labels would be useful for users to get oriented. I also think map colors should be consistent across the site, but I'm happy to change the others.

**Comment by beamalsky** [comment]
After deleting the preset CensusAreas and rerunning `import_data` this worked beautifully! One small thing: could you adjust the DataTables filter settings so you can filter by preset but not deactivate?

Screen Shot 2019-06-25 at 2 32 29 PM


 Lmk if you need a link to where that's happening in the template.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeancochrane) [reply README.md]: Great! Added the link.

File: surveys/templates/partials/multiselect_geometry_widget.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeancochrane) [reply surveys/templates/partials/multiselect_geometry_widget.html]: Nice, I didn't know you could set default map styles for `django-leaflet`, that's neat. I tweaked the styles a little bit to set the zoom/center around all of Philly and use the Carto Voyager basemap, which I think looks nice for both CensusAreas and Location/StudyArea selection:

Screen Shot 2019-06-25 at 3 12 13 PM
Screen Shot 2019-06-25 at 3 12 39 PM

The default OSM base tiles did feel too busy to me, but I'm curious whether you agree:

Screen Shot 2019-06-25 at 3 06 40 PM

**Comment by jeancochrane** [comment]
> could you adjust the DataTables filter settings so you can filter by preset but not deactivate?

Great catch, I tweaked this in 634b6d2!

File: surveys/templates/partials/multiselect_geometry_widget.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (beamalsky) [reply surveys/templates/partials/multiselect_geometry_widget.html]: Your styling changes as they currently stand in this PR look great to me! Thanks for tweaking the centering.

**Pull request merged**